airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java

```diff
@@ -77,17 +81,30 @@ public int hashCode() {
 
   public static MssqlCdcTargetPosition getTargetPosition(final JdbcDatabase database, final String dbName) {
     try {
-      final List<JsonNode> jsonNodes = database
-          .bufferedResultSetQuery(connection -> connection.createStatement().executeQuery(
-              "USE [" + dbName + "]; SELECT sys.fn_cdc_get_max_lsn() AS max_lsn;"), JdbcUtils.getDefaultSourceOperations()::rowToJson);
+      // We might have to wait a bit before querying the max_lsn to give the CDC capture job
+      // a chance to catch up. This is important in tests, where reads might occur in quick succession
+      // which might leave the CT tables (which Debezium consumes) in a stale state.
+      final JsonNode sourceConfig = database.getSourceConfig();
+      final Duration delay = (sourceConfig != null && sourceConfig.has("is_test") && sourceConfig.get("is_test").asBoolean())
+          ? MAX_LSN_QUERY_DELAY_TEST
```

Comment:
maybe we should provide that as a parameter to set when creating the source. Different customers might have that set to different values on their source DB

Reply:
if you agree, we should just create a github ticket. I'm not asking for that to be part of this change

Reply:
I'd like to put this off until we see some actual production usage. The default interval is 5 seconds. Presumably, airbyte syncs don't run that frequently, in which case this is irrelevant. Let's not anticipate problems which may never materialize.